src/preferences/broadcastsettings.h

```diff
@@ -1,97 +1,39 @@
-    QString getDefaultHost() const;
-    bool getLimitReconnects() const;
-    void setLimitReconnects(bool value);
-    bool getDefaultLimitReconnects() const;
-    QString getLogin() const;
-    void setLogin(const QString& value);
-    QString getDefaultLogin() const;
-    int getMaximumRetries() const;
-    void setMaximumRetries(int value);
-    int getDefaultMaximumRetries() const;
-    QString getMetadataCharset() const;
-    void setMetadataCharset(const QString& value);
-    QString getDefaultMetadataCharset() const;
-    QString getMetadataFormat() const;
-    void setMetadataFormat(const QString& value);
-    QString getDefaultMetadataFormat() const;
-    QString getMountpoint() const;
-    void setMountPoint(const QString& value);
-    QString getDefaultMountpoint() const;
-    bool getNoDelayFirstReconnect() const;
-    void setNoDelayFirstReconnect(bool value);
-    bool getDefaultNoDelayFirstReconnect() const;
-    bool getOggDynamicUpdate() const;
-    void setOggDynamicUpdate(bool value);
-    bool getDefaultOggDynamicUpdate() const;
-    QString getPassword() const;
-    void setPassword(const QString& value);
-    QString getDefaultPassword() const;
-    int getPort() const;
-    void setPort(int value);
-    int getDefaultPort() const;
-    double getReconnectFirstDelay() const;
-    void setReconnectFirstDelay(double value);
-    double getDefaultReconnectFirstDelay() const;
-    double getReconnectPeriod() const;
-    void setReconnectPeriod(double value);
-    double getDefaultReconnectPeriod() const;
-    QString getServertype() const;
-    void setServertype(const QString& value);
-    QString getDefaultServertype() const;
-    QString getStreamDesc() const;
-    void setStreamDesc(const QString& value);
-    QString getDefaultStreamDesc() const;
-    QString getStreamGenre() const;
-    void setStreamGenre(const QString& value);
-    QString getDefaultStreamGenre() const;
-    QString getStreamName() const;
-    void setStreamName(const QString& value);
-    QString getDefaultStreamName() const;
-    bool getStreamPublic() const;
-    void setStreamPublic(bool value);
-    bool getDefaultStreamPublic() const;
-    QString getStreamWebsite() const;
-    void setStreamWebsite(const QString& value);
-    QString getDefaultStreamWebsite() const;
+    void setCurrentProfile(BroadcastProfile* profile);
+    BroadcastProfile* getCurrentProfile();
+    BroadcastProfile* getProfileByName(const QString& profileName);
+    void saveProfile(BroadcastProfile* profile);
+    void saveAll();
+    void deleteProfile(BroadcastProfile* profile);
 
   private:
+    void loadProfiles();
+    QString filenameForProfile(BroadcastProfile* profile);
+    QString filenameForProfile(const QString& profileName);
+    QString getProfilesFolder();
+    void loadLegacySettings(BroadcastProfile* profile);
+
     // Pointer to config object
     UserSettingsPointer m_pConfig;
+    QList<BroadcastProfile*> m_profiles;
```

Comment:
Fixed, I replaced the `QList<BroadcastProfile*` with an `std::vector<std::unique_ptr<BroadcastProfile>>`.